Sources/StreamChatUI/Composer/ComposerVC.swift

```diff
@@ -995,6 +1025,67 @@ open class ComposerVC: _ViewController,
         NotificationCenter.default.post(name: .createNewPoll, object: nil, userInfo: userInfo)
     }
 
+    @objc func sendWeatherMessage(_ notification: Notification ) {
+        DispatchQueue.main.asyncAfter(deadline: .now() + 0.2) { [weak self] in
+            guard let `self` = self else { return }
+            self.channelController?
+                .createNewMessage(
+                    text: "/weather",
+                    extraData: ["weather": .dictionary(self.getWeatherRequest(notification))],
+                    completion: nil
+                )
+        }
+    }
+
+    @objc func editWeatherMessage(_ notification: Notification ) {
+        let messageId = notification.userInfo?["messageId"] as? String ?? ""
+        guard let cid = channelController?.cid else { return }
+        let action = AttachmentAction(name: "action", value: "edit", style: .default, type: .button, text: "Edit")
+        ChatClient.shared.messageController(cid: cid, messageId: messageId)
+            .dispatchEphemeralMessageAction(action, getWeatherRequest(notification), key: "weather")
+    }
+
+    func getWeatherRequest(_ notification: Notification) -> [String: RawJSON] {
+        var weatherData = [String: RawJSON]()
+        // Get data from notification
+        let temp = notification.userInfo?["tempreature"] as? String ?? ""
+        let location = notification.userInfo?["location"] as? String ?? ""
```

Comment:
@mdhanif-simformsolutions  get userInfo in separate variable and use it instead.